### DISCUSSION TITLE
Any thoughts about keeping Tidal and Deezer going without mysqueezebox?

### DISCUSSION THREAD

**Discussion by baslking** [LMS-Community/slimserver#994] [General]
Did anyone get warning that this was coming?
https://www.mysqueezebox.com/index/Home

**Comment by michaelherger** [comment]
See https://forums.slimdevices.com/node/1668327 and related threads. This actually _is_ the warning...

**Comment by anotherstuart** [comment]
Unless there is something that I have misunderstood there must be a lot of reusable code in the 'Home Assistant' Music Assistant project which has deezer up and running with a proper Oauth login. Disucssions https://github.com/orgs/music-assistant/discussions/1245 and https://github.com/music-assistant/server/pull/625

I will, over time, have a look at this but I have little coding experience in Python and even less in making LMS plugins.

Anyone else that is willing to pick this up and I will help if I can. Would be interested to know what @arctixdev and @micha91 who developed the suff for Musica Assistant think about it.

**Comment by philippe44** [comment]
One is Python and using 3rd party libraries, the other is Perl and have to be re-written from scratch. So other than "knowing what to do" (which is very important of course) but there is no reuse of code at all.

**Reply by anotherstuart** [reply]
Thanks @philippe44 , confirming my lack of knowledge of LMS plugins!  : )

One more naïve question - if you can indulge me - is the current implementation of Deezer largely embedded in the LMS code / plugin (rather than at the mysqueezebox end) to the extent that implementing the Oauth login parts and some fiddling would suffice, or is it a matter of more or less coding a complete Deezer plugin from the ground up?

**Reply by philippe44** [reply]
I suspect this is the same as the Tidal plugin (I would have to check) where it's basically a re-coding with limited reuse
